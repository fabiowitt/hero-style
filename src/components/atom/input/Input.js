import React from 'react';
import classNames from 'classnames';
import css from './Input.module.scss';

const Input = (props) => {
  const {disabled, type} = props;
  const isCheckOrRadio = type==='checkbox' || type==='radio';

  const classes = classNames({
    [css['c-input']]: true,
    [css['c-input--disabled']]: disabled,
    [css[`c-input__${type}`]]: isCheckOrRadio,
    [css[`c-input__${type}--disabled`]]: disabled && isCheckOrRadio
  });

  return <input
    className={classes}
    {...props}
  />
};

Input.defaultProps = {
  type: 'text'
};

export default Input;
