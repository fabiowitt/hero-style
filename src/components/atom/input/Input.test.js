import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('Input', () => {
  it('should match with the snapshot', () => {
    const wrapper = renderer
      .create(<Input placeholder={'placeholder text'} id={'ipt'} name={'ipt'} type={'text'}></Input>)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});