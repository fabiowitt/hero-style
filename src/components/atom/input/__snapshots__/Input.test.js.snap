// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match with the snapshot 1`] = `
<input
  className="c-input"
  disabled={null}
  id="ipt"
  name="ipt"
  onChange={[Function]}
  placeholder="placeholder text"
  type="text"
  value={null}
/>
`;
